cookbooks/ros2_windows/recipes/rti_connext.rb

```diff
@@ -0,0 +1,118 @@
+connext_params = node['ros2_windows']['rti_connext']
+
+# These will fail if the rti_connext parameters have not been specified because the defaults are 'nil'
+assert(File.exists?(connext_params['license_file']),
+                    "License file location does not exist: #{connext_params['license_file']}")
+
+assert(Dir.exists?(connext_params['installer_dir']),
+                   "Installer directory does not exist: #{connext_params['installer_dir']}")                          ))
+assert_not_nil(connext_params['version'], "Version is nil, requires MAJOR.MINOR.PATCH (e.g. '5.3.1')")
+assert_not_nil(connext_params['edition'], "Edition is nil, requires one of ('evaluation', 'pro')")
+
+assert_not_nil(connext_params['min_vs_version'], "Minimum Visual Studio version is required (e.g. '2017')")
+
+assert_not_nil(connext_params['openssl_version'], "OpenSSL version is required (e.g. '1.0.2n')")
```

Comment:
Okay I think what we want to do here is use the Chef logging facility to log fatal errors and then raise an error. The log resource has some examples: https://docs.chef.io/resources/log/